### DISCUSSION TITLE
Devices in app but not in API

### DISCUSSION THREAD

**Discussion by neuhausf** [astrandb/miele#246] [Q&A] [answered]
First: The API-access works, but I get no devices listed. Although I have 2 devices in my app.

Problems seems to be:
miele Website Login != App Login

* I created an account at miele.ch *website* and tried to login in the app:
app account doesn't exist. So I used *the same e-mail* and registered in the app (iOS). Now I can login and add my devices. But the devices do not get listed through the API!
* I also tried the other way round (app account first and then website account). Note that the API login works only, if created an account through the site.
* I think the two accounts are not linked. Is this a problem with swiss accounts only? Should I register through another country? Any ideas?

**Comment by astrandb** [comment]
You can test your accounts on the developer site https://www.miele.com/developer/swagger-ui/ to bring clarity on what is going on.

**Reply by neuhausf** [reply]
Thanks for the suggestion. That's what I did. After creating an account through the website and ordering API credentials, I can login in the swagger. I used the endpoint /devices to test what's coming - and there is just an empty response: {}
In the iOS app I see 2 devices, which I can control.

**Reply by astrandb** [reply]
It seems that you have got two disconnected accounts. You are the first user of the integration that reports this problem. I suggest that you contact Miele in order to get their help to sort it out.

**Reply by neuhausf** [reply]
Thanks @astrandb. I was trying three different e-mail-addresses and moved my two devices everytime. To be sure to make things right, is this the process you suggest?
1. use the iOS/Android app to setup an account
2. register your devices through the app
3. go to developer-portal and apply for an API-access with the used e-mail in 1)
4. no actions through the website (account registering at miele.ch or whatever country) required?

What country are you in? I want to try another one (if I can override this in the miele app...).

**Reply by astrandb** [reply]
I am in Sweden. If I remember correctly I did things in same order as you describe.

**Reply by neuhausf** [reply, marked as answer]
Ok - problem was sitting in front of the computer (or iPhone in this case)...
There are really different accounts per country. On my iPhone, the preselected country was the topmost ("Afghanistan" or something like this). Probably because I'm using an AdGuard DNS/VPN. So the account was not created in the expected country.
